analyzer/evmtokens/evm_tokens.go

```diff
@@ -79,7 +80,13 @@ type StaleToken struct {
 
 func (p *processor) GetItems(ctx context.Context, limit uint64) ([]*StaleToken, error) {
 	var staleTokens []*StaleToken
-	rows, err := p.target.Query(ctx, queries.RuntimeEVMTokenAnalysisStale, p.runtime, limit)
+	// Use strict isolation; cannot afford non-repeatable reads where the total supply and the download round would be read
```

Comment:
serializable good :smiling_face_with_tear: from the explanation. I think the thing that has me all weird is that it doesn't particularly need repeatable reads, only that two reads for different things (height H and balance N) need to be consistent

Reply:
or is it, docs say

> Read Committed is the default isolation level in PostgreSQL. When a transaction uses this isolation level, a SELECT query (without a FOR UPDATE/SHARE clause) sees only data committed before the query began; it never sees either uncommitted data or changes committed by concurrent transactions during the query's execution. In effect, a SELECT query sees a snapshot of the database as of the instant the query begins to run.

which should prevent the later read of N from giving the data from height H+1

Reply:
I don't know how to consolidate your snippet ([from here](https://www.postgresql.org/docs/current/transaction-iso.html#XACT-READ-COMMITTED)):

> In effect, a SELECT query sees a snapshot of the database as of the instant the query begins to run.

with [this](https://www.postgresql.org/docs/current/transaction-iso.html):

> nonrepeatable read: A transaction re-reads data it has previously read and finds that data has been modified by another transaction (that committed since the initial read).
> (later in the table: In Read Committed mode, nonrepeatable reads are possible.)

Edit: Oh ... I see. Your link says  (emphasis added)

> In effect, a SELECT query sees a snapshot of the database as of the instant the **query** begins to run.

and also

> Also note that two successive SELECT commands can see different data, even though they are within a single transaction, if other transactions commit changes after the first SELECT starts and before the second SELECT starts.

This was all part of the slack discussion. I first thought that Read Committed ensured consistency within a single SELECT, then convinced myself that it does not (and started this PR), but now I see that was wrong. Stupid subtlety :shakefist:.

I'll close this PR, I think it doesn't really make sense any more. Thanks for the pushback.